src/lib/metrics.ts

```diff
@@ -714,367 +1062,56 @@ export default class MetricsMonitor {
                         config.rateLimiting.callSignalEndpointMaxPerSecond * 60,
                     );
             } catch (e) {}
-        }
-
-        await schedulerService.schedule(
-            collectStaticCounters.bind(this),
-            hoursToMilliseconds(2),
-            'collectStaticCounters',
-            0, // no jitter
-        );
-
-        eventBus.on(
-            events.REQUEST_TIME,
-            ({ path, method, time, statusCode, appName }) => {
-                requestDuration
-                    .labels({
-                        path,
-                        method,
-                        status: statusCode,
-                        appName,
-                    })
-                    .observe(time);
-            },
-        );
-
-        eventBus.on(events.SCHEDULER_JOB_TIME, ({ jobId, time }) => {
-            schedulerDuration.labels(jobId).observe(time);
-        });
-
-        eventBus.on(
-            events.FUNCTION_TIME,
-            ({ functionName, className, time }) => {
-                functionDuration
-                    .labels({
-                        functionName,
-                        className,
-                    })
-                    .observe(time);
-            },
-        );
-
-        eventBus.on(events.EVENTS_CREATED_BY_PROCESSED, ({ updated }) => {
-            eventCreatedByMigration.inc(updated);
-        });
-
-        eventBus.on(events.FEATURES_CREATED_BY_PROCESSED, ({ updated }) => {
-            featureCreatedByMigration.inc(updated);
-        });
-
-        eventBus.on(events.DB_TIME, ({ store, action, time }) => {
-            dbDuration
-                .labels({
-                    store,
-                    action,
-                })
-                .observe(time);
-        });
+        },
+    };
+}
+export default class MetricsMonitor {
+    constructor() {}
 
-        eventBus.on(events.PROXY_REPOSITORY_CREATED, () => {
-            proxyRepositoriesCreated.inc();
-        });
+    async startMonitoring(
+        config: IUnleashConfig,
+        stores: IUnleashStores,
+        version: string,
+        eventBus: EventEmitter,
+        instanceStatsService: InstanceStatsService,
+        schedulerService: SchedulerService,
+        db: Knex,
+    ): Promise<void> {
+        if (!config.server.serverMetrics) {
+            return Promise.resolve();
+        }
 
-        eventBus.on(events.FRONTEND_API_REPOSITORY_CREATED, () => {
-            frontendApiRepositoriesCreated.inc();
-        });
+        collectDefaultMetrics();
 
-        eventBus.on(events.PROXY_FEATURES_FOR_TOKEN_TIME, ({ duration }) => {
-            mapFeaturesForClientDuration.observe(duration);
-        });
+        const { collectStaticCounters, collectDbMetrics } =
+            registerPrometheusMetrics(
+                config,
+                stores,
+                version,
+                eventBus,
+                instanceStatsService,
+            );
```

Comment:
Start monitoring now calls helper functions that register prometheus metrics and then adds schedules to frequently update these metrics